trillian/integration/chains.go

```diff
@@ -0,0 +1,245 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integration
+
+import (
+	"crypto"
+	"crypto/rand"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/google/certificate-transparency-go/x509"
+	"github.com/google/trillian/crypto/keys"
+	"github.com/google/trillian/crypto/keyspb"
+
+	ct "github.com/google/certificate-transparency-go"
+)
+
+// ChainGenerator encapsulates objects that can generate certificate chains for testing.
+type ChainGenerator interface {
```

Comment:
Might have used Producer or similar here instead of Generator but not going to really argue it. It's a source of certificates, not necessarily new ones.